tests/test_workers/test_tasks/test_gomod.py

```diff
@@ -52,10 +99,11 @@ def test_fetch_gomod_source(
         "module_deps": sample_deps_replace,
         "packages": [{"pkg": sample_pkg_lvl_pkg, "pkg_deps": pkg_lvl_deps}],
     }
-    tasks.fetch_gomod_source(1, dep_replacements)
+    mock_bundle_dir.return_value.app_subpath.side_effect = directory_present
```

Comment:
I didn't know you could use `side_effect` like this! :rocket:

Reply:
yeah... `side_effect` is super powerful! The challenge is to do it in a way that does not increase the code complexity _too_ much.